tests/integration/features/bootstrap/FeatureContext.php

```diff
@@ -1487,6 +1489,18 @@ public function userBansUserFromRoom(string $user, string $actorType, string $ac
 			}
 		}
 
+		//Users or Moderators can't ban themselves
+		if($user === $actorId) {
+			return;
+		}
+
+		//If actorId is a moderator
+		if(isset(self::$promotedParticipants[$identifier])) {
+			if($actorId === self::$promotedParticipants[$identifier]) {
+				return;
+			}
+		}
```

Comment:
this must not be here. You should block this in the actual BanController/BanService